packages/components/src/ListBox/src/ListBoxItem.tsx

```diff
@@ -0,0 +1,313 @@
+                            [DEFAULT_SLOT]: {
+                                className: styles["hop-ListBoxItem__icon-list"],
+                                size: ListBoxItemToIconSizeAdapter[size]
+                            },
+                            "end-icon": {
+                                className: styles["hop-ListBoxItem__end-icon-list"],
+                                size: ListBoxItemToIconSizeAdapter[size]
+                            }
+                        }
+                    }],
+                    [IconContext, {
+                        slots: {
+                            [DEFAULT_SLOT]: {
+                                className: styles["hop-ListBoxItem__icon"],
+                                size: ListBoxItemToIconSizeAdapter[size]
+                            },
+                            "end-icon": {
+                                className: styles["hop-ListBoxItem__end-icon"],
+                                size: ListBoxItemToIconSizeAdapter[size]
+                            }
+                        }
+                    }],
+                    [BadgeContext, {
+                        className: styles["hop-ListBoxItem__badge"],
+                        isDisabled: isDisabled
+                    }]
+                ]}
+            >
+                {children}
+            </SlotProvider>
+        </>
+    );
+}
+
+function ListBoxItem<T extends object>(props: ListBoxItemProps<T>, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, ListBoxItemContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children: childrenProp,
+        isInvalid,
+        size: sizeProp,
+        style: styleProp,
+        textValue: textValueProp,
+        selectionIndicator,
+        ...otherProps
+    } = ownProps;
+
+    const size = useResponsiveValue(sizeProp) ?? "sm";
+    const textValue = textValueProp ?? (typeof childrenProp === "string" ? childrenProp : undefined);
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalListBoxItemCssSelector,
+        cssModule(
+            styles,
+            "hop-ListBoxItem",
+            size,
+            selectionIndicator
+        ),
+        stylingProps.className
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const children = composeRenderProps(childrenProp, prev => {
+        if (prev && isTextOnlyChildren(prev)) {
+            return <Text slot="label">{prev}</Text>;
+        }
+
+        return prev;
+    });
+
+    const [isListHasSelection, setIsListHasSelection] = useState(false);
+    const [isListHasSelectionEnd, setIsListHasSelectionEnd] = useState(getIsListHasSelectionEnd(isListHasSelection, false));
+
+    const handleTransitionEnd = useCallback((event: TransitionEvent) => {
```

Comment:
can you extract a hook from this? i feel like RAC did a good job with their useExitAnimation, can you combine what you have with what they did?

Reply:
well since I re-added the fic, I'm able to put the onTransitionEnd directly on the div since it's not a RAC element. i think this is much simpler. We can always add a hook later on if it comes up again.